cypress/integration/select_server.spec.ts

```diff
@@ -0,0 +1,77 @@
+describe('Select Server Page Tests', () => {
+  it('Redirects to Select Server', () => {
+    cy.visit('http://localhost:3000');
+
+    cy.url().should('include', '/selectserver');
+  });
+
+  it('Connect Button is visible but disabled', () => {
+    cy.visit('http://localhost:3000/selectserver');
+
+    // Buttons should contain the text "connect"
+    cy.get('form')
+      .get('button[type="submit"]')
+      .contains('connect', { matchCase: false });
+
+    // Since the text field is blank, the form buttons should be disabled
+    cy.get('form').get('button[type="submit"]').should('be.disabled');
+  });
+
+  it('Server input is vislble, and contains label', () => {
```

Comment:
This is more akin to unit tests for the layout than to real integration tests.

Imo, the process of testing this page should be:

* Load the page
* Input is present
* Connect button is disabled
* Entering an URL in the input enables the button
* Click on the button
* It works, test passes

Then another identical test with an invalid URL, to test validation.

Reply:
The idea here is to start from a blank slate for every situation, to ensure things are clean, then test the full path the user would take to reach that situation.

It works pretty much like "user stories" work in project management, you describe the process the user does and what the expected result is.